cocos/core/math/mat3.ts

```diff
@@ -793,16 +790,17 @@ export class Mat3 extends ValueType {
     public set (m00: number | Mat3 = 1, m01 = 0, m02 = 0,
         m03 = 0, m04 = 1, m05 = 0,
         m06 = 0, m07 = 0, m08 = 1): Mat3 {
+        const self = this;
```

Comment:
The same as above.

Reply:
For generating smaller code.

`this.` could not be transformed to shorter code.
If there are lots of `this.` in a funciton, we define a temporary variable `self` for them, so all `self.xxx` will be minified to shorter code like `a.xxx`

Reply:
Got it. We should write a documentation for it. There are too many limitations.

Reply:
I will write the best practice documentation later for smaller code size.
BTW, it's not a limitation here, it just a best practice.